pkg/agent/informant.go

```diff
@@ -746,6 +751,17 @@ func (s *InformantServer) handleSuspend(ctx context.Context, body *api.SuspendAg
 		return nil, 400, errors.New("Cannot suspend agent that is not yet registered")
 	}
 
+	// Acquire s.runner.requestLock so that when we return, we can guarantee that
+	locked = false
+	s.runner.lock.Unlock()
```

Comment:
having two different locks within one runner struct is definitely a bit suspicious and difficult to get my head around. (and I think is one of the patterns that we would want to get rid of in #34. 

The comment doesn't help as much as it might?